server/watttime.go

```diff
@@ -413,3 +415,20 @@ func staticGetWattTimeToken(username, password string) (string, error) {
 
 	return tokenResponse.Token, nil
 }
+
+// threadedGetWattTimeWeekData wakes up periodically and refreshes the weekly
+// WattTime data.
+func (gcas *GCAServer) threadedGetWattTimeWeekData(username, password string) {
+	for {
+		select {
+		case <-gcas.quit:
+			return
+		case <-time.After(WattTimeWeekDataUpdateFrequency):
+		}
+
+		err := gcas.managedGetWattTimeWeekData(username, password)
+		if err != nil {
+			gcas.logger.Errorf("Threaded call unable to get WattTime data for the most recent week: %v", err)
+		}
```

Comment:
Should add a comment here that specifies that the function is called during startup, therefore it's safe to sleep before calling it here.